server/controllers/request.js

```diff
@@ -6,13 +6,15 @@ const mongoose = require("mongoose");
 // @desc list of requests for logged in user
 // @access Private
 exports.getRequests = asyncHandler(async (req, res, next) => {
-  const userId = req.body.userId;
+  const userId = req.query.userId;
```

Comment:
The `id` of the logged in user will be available in `req.user.id` hence there is no need to use this query parameter.

NOTE: If you wanted/needed to use the `id` parameter for sure, I'd recommend using path parameters instead ([example](https://stackoverflow.com/a/31261026)).